### PR TITLE
Improved vault-show error message

### PR DESCRIPTION
**Pull request by stlaz**
Added more information to the NotFound error that may occur during
execution of vault-show. It was not clear whether the vault
really does not exist or it does not exist in a certain container.

https://fedorahosted.org/freeipa/ticket/5950

### CODE REVIEW

**Comment by mbasti-rh** [comment]
NACK: you fixed only vault-show not other vault-* commands

NACK: I don't like the override of execute method (it should work for all vault-* commands automatically)

Is possible to override method handle_not_found of vault object? IMO which vault type is used can be determined by DN suffix (maybe it deserves a new method vault_type_from_DN()).

NACK: This is pure evil, pls keep better readability (use if-elif-else instead)
```
            if options.get('service'):
                container_type = 'service'
            else:
                container_type = 'shared' if options.get('shared') else 'user'
```

**Comment by mbasti-rh** [comment]
Oh realized that is not possible to create DN inside handle_not_found, because it does not take **kwargs

Probably we can extend handle_not_found with kwargs, but this decision needs broader audience.

**Comment by stlaz** [comment]
Seems like nobody objected so far.

File: ipaserver/plugins/vault.py [hunk 1]
```diff
@@ -755,6 +755,28 @@ def get_container_attribute(self, entry, options):
         elif entry.dn.endswith(DN(('cn', 'users'), container_dn)):
             entry['username'] = entry.dn[1]['cn']
 
+    def handle_not_found(self, *keys, **kwargs):
```
Comment (stlaz) [inline ipaserver/plugins/vault.py]: I suppose you're right, fixed it here.

**Comment by mbasti-rh** [comment]
I had discussion with @jcholast and he disagrees. This weird handling of DN should stay isolated in vault code and shouldn't be spreaded across the framework. I'm starting to think that we should close ticket as won't/can't fix instead of doing that bad code even worse.

**Comment by stlaz** [comment]
WONTFIX then. There's no winning here.

**Pull request closed without merging**